### ISSUE TITLE
X509 cert.sign() method always signs using SHA1withRSA

### ISSUE THREAD

**Issue by hefloryd** [kjur/jsrsasign#241]
In https://github.com/kjur/jsrsasign/blob/a3185663042c971aed5ef5b4bacd7e9f55711e5a/asn1x509-1.0.js#L163, the certificate is always signed using SHA1withRSA regardless of the chosen signature algorithm.

If e.g. SHA256withRSA is chosen, the resulting certificate is not valid in Chrome or IE (but accepted by Firefox funnily enough).

The following code was used to generate a SHA256withRSA certificate that works on all tested browsers:

```
      var kp = KEYUTIL.generateKeypair("RSA", 1024);

      var o = new KJUR.asn1.x509.TBSCertificate();
      o.setSerialNumberByParam({'int': 99});
      o.setSignatureAlgByParam({'name': 'SHA256withRSA', paramempty: false}); // NOTE: paramempty: false to get NULL in AlgorithmIdentifier (possibly not required)
      o.setIssuerByParam({'str': '/CN=192.168.10.103'});
      o.setNotBeforeByParam({'str': '170306134420Z'});
      o.setNotAfterByParam({'str': '170405134420Z'});
      o.setSubjectByParam({'str': '/CN=192.168.10.103'});
      o.setSubjectPublicKeyByGetKey(kp.pubKeyObj);
      o.appendExtensionByName('BasicConstraints', {cA: false});
      o.appendExtensionByName('ExtKeyUsage', {array: [{name: "serverAuth"}]});

      // Sign cert (this will always use SHA1withRSA)
      // var cert = new KJUR.asn1.x509.Certificate({'tbscertobj': o, 'prvkeyobj': kp.prvKeyObj});
      // cert.sign();

      // Sign using SHA256withRSA
      var cert = new KJUR.asn1.x509.Certificate({'tbscertobj': o, 'prvkeyobj': kp.prvKeyObj});
      var sig = new KJUR.crypto.Signature({'alg': 'SHA256withRSA'});
      sig.init(kp.prvKeyObj);
      sig.updateHex(o.getEncodedHex());
      var hexSig = sig.sign();
      cert.setSignatureHex(hexSig);

      var certPEM = cert.getPEMString();
```

**Comment by kjur** [comment]
Thank you for your report. I'll check it later.

**Comment by kjur** [comment]
This issue was fixed in 7.1.1 release today. Thanks.
